apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -103,17 +105,23 @@ export default class AssignmentSelector extends Component {
     this.setState({
       selectedPrimaryId,
       selectedSecondaryId
-    });
+    }, this.reportChange);
   };
 
   onChangeSecondary = (event) => {
     this.setState({
       selectedSecondaryId: event.target.value
-    });
+    }, this.reportChange);
+  };
+
+  reportChange = () => {
+    if (typeof this.props.onChange === 'function') {
+      this.props.onChange(this.getSelectedAssignment());
```

Comment:
Presumably this makes it so that we dont have to have other components reach into this one and ask for `selectedAssignment` now? :)